reviews/dbsc-analysis.md

```diff
@@ -0,0 +1,158 @@
+
+## Signed Cookies
+
+That design might include a new `Signed` parameter for cookies in `Set-Cookie`. That attribute would request that, whenever the cookie is sent to the server, the client would cover that cookie with a signature.
+
+The use of the `Path` cookie parameter would ensure that `Signed` cookies are only sent when requests are made to specific resources. That would allow servers to limit how often they ask clients to generate signatures by limiting how often the client requests the identified resources. Those requests could be initiated through a redirect, fetch, or any other request, as needed.
+
+The choice of what fields to include under the signature is very important.  It is not likely to be sufficient to just cover `Cookie` in the list of signed content in `Signature-Input`. Including a date (the `created` parameter), the method (`@method`), and the URL (`@target-uri`) seem to be the minimum set of things that will prevent the signature from being reused, but it's possible that a more thorough security analysis will notice some other fields that need to be covered in order to be secure.
+
+## Some Challenges
+
+A potential challenge then is coordinating those requests so that different origins within the site, which might be only loosely coordinated through a central authentication/authorization system, don’t generate more requests too often. That can be managed by directing refresh requests to a resource on that system that does not have `Signed` cookies associated with it. That resource can coordinate any cookie refreshes, forwarding requests to the affected path as necessary.
+
+Another challenge is in demonstrating liveness for the signature. TPMs don’t generally have clocks, so if a device is compromised so that an attacker gains access to the TPM, the attacker could generate an arbitrary number of signatures for future use. However, this requires that the attacker predict the times and URLs where those signatures would be needed. This suggests a similar pattern to solve that potential problem also: the server redirects to a new endpoint with fresh randomness in the URL for signing.
+
+For example, both requirements could be addressed as shown below.  This example is expanded to include the maximum number of exchanges possible to fully illustrate all of the capabilities.
+
+```http
+GET /some/resource
+Cookie: login=expired
+```
+
+Which results in a redirection to a login endpoint, as would be part of a normal centralized login flow (i.e., this would be a perfectly normal part of refreshing cookies):
+
+```http
+303 Over Yonder
+Location: /login
+```
+
+Consequently, the client follows the redirect.
+
+```http
+GET /login
+Cookie: login=expired
+```
+
+This resource then makes a call about the freshness of the login cookies and determines that a signature is needed.  It initiates another redirect, including fresh entropy in the URL to guarantee that the client has live TPM access.
+
+```http
+303 Over There Now
```

Comment:
The custom reason-phrases are distracting me, and making me wonder if they have a semantic meaning. They [don't](https://httpwg.org/specs/rfc9112.html#status.line), but we should avoid distractions for other readers.